vercel-bot/api/index.ts

```diff
@@ -3,6 +3,11 @@ import { Bot, webhookCallback } from 'grammy'
 
 const bot = new Bot(process.env['BOT_TOKEN'])
 
-export default async (req, res) => { // <-- Fill in the address of your deployed vercel allocation.
-  webhookCallback(bot, 'http')(req, res) // <-- The free version of vercel has restrictions on quotas, which we need to enable in the configuration file vercel.json
-}
\ No newline at end of file
+// attach all middleware
+bot.on("message", async (ctx) => {
+  await ctx.reply("Hi there!")
+})
+
+// The free version of vercel has restrictions on quotas, which we need to enable in the configuration file vercel.json
+// webhookCallback will make sure that the correct middleware(listener) function is called
+export default webhookCallback(bot, 'express')(req, res)
```

Comment:
Sorry, I missed this.
```suggestion
export default webhookCallback(bot, 'express')
```

Reply:
🤦